Improve `Object.get_property_list()` method description

mateuseap opened the pull request:
## What I did 

* Improved the `Object.get_property_list()` method description by making more clear what are the properties returned.
* Closes: godotengine/godot-docs#7533

_Bugsquad edit:_ Fixed method name and formatting.

dalexeev commented:
This is not correct in general, for example in GDScript `get_property_list()` returns all variables:

```gdscript
var a: int
@export var b: int

func _ready() -> void:
    for p in get_property_list():
        print(p)
```

```
...
{ "name": "a", "class_name": &"", "type": 2, "hint": 0, "hint_string": "", "usage": 4096 }
{ "name": "b", "class_name": &"", "type": 2, "hint": 0, "hint_string": "int", "usage": 4102 }
```

I think it should be a note at the end of the method description explicitly mentioning C#.

mateuseap commented:
> [godotengine/godot-docs#7533](https://github.com/godotengine/godot-docs/issues/7533):
> 
> > Godot property is different from a C# property (not sure if "property" is a concept in gdscript that could also cause confusion). Would be great if this was clarified in the docs.
> 
> This is not correct in general, for example in GDScript `get_property_list()` returns all variables:
> 
> ```gdscript
> var a: int
> @export var b: int
> 
> func _ready() -> void:
>     for p in get_property_list():
>         print(p)
> ```
> 
> ```
> ...
> { "name": "a", "class_name": &"", "type": 2, "hint": 0, "hint_string": "", "usage": 4096 }
> { "name": "b", "class_name": &"", "type": 2, "hint": 0, "hint_string": "int", "usage": 4102 }
> ```
> 
> I think it should be a note at the end of the method description explicitly mentioning C#.
> 
> @mateuseap Please update also the commit message.

@dalexeev thanks for helping me! I removed my previous modifications, added a note in the end of `get_property_list()` method description and adjusted the commit message. I hope that's ok, if it's not, I'm open to modify it. Again, thanks for the help! :)

YuriSizov commented:
I'm not sure what the added note is saying, and so I feel that it's wrong. It is true that while in GDScript all class members would be properties, in C# properties need to be explicitly opted in. But I'm not confident the added note helps.

The list of properties is not related to scripting languages. It's an overall engine concept. Built-in objects also contain properties and this method returns all of them. Properties can be added dynamically by overriding `_get_property_list` on your object too. This is true for any language.

I think that the note could simply state that C# and 3rd party languages may require you to explicitly mark class members as Godot properties.

mateuseap commented:
> I'm not sure what the added note is saying, and so I feel that it's wrong. It is true that while in GDScript all class members would be properties, in C# properties need to be explicitly opted in. But I'm not confident the added note helps.
> 
> The list of properties is not related to scripting languages. It's an overall engine concept. Built-in objects also contain properties and this method returns all of them. Properties can be added dynamically by overriding `_get_property_list` on your object too. This is true for any language.
> 
> I think that the note could simply state that C# and 3rd party languages may require you to explicitly mark class members as Godot properties.

@YuriSizov done! I still mantained the statement that says that in GDScript all class members are treated as properties, but I've clarified that in C# and 3rd party languages may require you to explicitly mark class members as Godot properties, as you've pointed out.

Calinou commented on doc/classes/Object.xml:
```diff
@@ -651,6 +651,7 @@
 				- [code]hint[/code] is [i]how[/i] the property is meant to be edited (see [enum PropertyHint]);
 				- [code]hint_string[/code] depends on the hint (see [enum PropertyHint]);
 				- [code]usage[/code] is a combination of [enum PropertyUsageFlags].
+				[b]Note:[/b] While in GDScript all class members are treated as properties, in C# and some 3rd party languages may be necessary to explicitly mark class members as Godot properties using decorators or attributes.
```
```suggestion
				[b]Note:[/b] In GDScript, all class members are treated as properties. In C# and GDExtension, it may be necessary to explicitly mark class members as Godot properties using decorators or attributes.
```

AThousandShips replied:
Should we clarify "member variables" to avoid any confusion?

mateuseap committed:
Improve Object.get_property_list() method description

YuriSizov reviewed (approved):
LGTM now!

akien-mga commented:
Thanks!
